docs/services/sdk/kafka.mdx

```diff
@@ -0,0 +1,235 @@
+
+You can mix event handlers with other event handlers or RPC handlers. For example, assume you're implementing a `profile` router to handle and store user profiles in Restate, where the key is the profile's unique identifier.
+In the same service, you can implement an RPC method called `get` to retrieve the profile, and you can attach an event handler called `updateProfile` to consume events from Kafka to update the profile.
+
+For each event key (= Kafka record key), the events are delivered in the order in which they arrived on the topic partition.
+
+</TabItem>
+<TabItem value="grpc" label="gRPC API">
+
+### Handling events with string key
+
+You can handle events within keyed services, where the key of the Kafka record matches the key used by the service:
+
+```protobuf
+syntax = "proto3";
+import "dev/restate/ext.proto";
+import "dev/restate/events.proto";
+
+service MyService {
+    option (dev.restate.ext.service_type) = KEYED;
+
+    rpc Handle(dev.restate.StringKeyedEvent) returns (google.protobuf.Empty);
+    // Other methods
+}
+```
+
+```typescript
+import * as restate from "@restatedev/restate-sdk";
+
+import MyService from "./generated/my_service";
+import StringKeyedEvent from "./generated/dev/restate/events";
+import Empty from "./generated/google/protobuf/empty";
+
+export class MyServiceImpl implements MyService {
+    async handleEvent(request: StringKeyedEvent): Promise<Empty> {
+      const ctx = restate.useContext(this);
+      const payload = request.payload;
+
+      // Handle event
+
+      return Empty.create({});
+    }
+
+    // Other service methods
+}
+```
+
+The Kafka record key must be a valid UTF-8 string, otherwise Restate won't be able to deliver the event to the service.
+
+You can mix methods receiving events with regular RPC methods. For example, assume you're implementing a `Profile` service to handle and store user profiles in Restate, where the key is the profile's unique identifier.
+In the same service, you can implement a  `Get` method to retrieve the profile, and you can attach an `UpdateProfile` event handler to consume events from Kafka to update the profile.
+
+For each event key (= Kafka record key), the events are delivered in the order in which they arrived on the topic partition.
+
+### Handling generic events from Kafka in order
+
+You can handle events where the Kafka record key is available but is not necessarily a string. To do so, you need to handle events in an ad-hoc service that process these events:
+
+For example:
+
+```protobuf
+syntax = "proto3";
+import "dev/restate/ext.proto";
+import "dev/restate/events.proto";
+
+service MyService {
+    option (dev.restate.ext.service_type) = KEYED;
+
+    rpc Handle(dev.restate.KeyedEvent) returns (google.protobuf.Empty);
+}
+```
+
+For each event key (= Kafka record key), thanks to Restate [`KEYED` service type](/services/service_type), the events are delivered in the order in which they arrived on the topic partition.
```

Comment:
For keyed services, events with the same key are delivered in the order in which they arrived on the topic partition.

Reply:
```suggestion
For keyed services, events with the same key are delivered in the order in which they arrived on the topic partition.
```